dor/providers/parsers.py

```diff
@@ -59,45 +64,54 @@ def get_data_files(self) -> list[FileMetadata]:
             for elem in self.tree.findall(".//METS:file")
         ]
 
-    def get_md_file_metadatum(self, elem):
+    def get_md_file_metadatum(self, elem: ElementAdapter):
         id_ = elem.get("ID")
         use = elem.get("USE")
-        locref = elem.find("METS:mdRef").get_optional("LOCREF")
-        mdtype = elem.find("METS:mdRef").get_optional("MDTYPE")
-        mimetype = elem.find("METS:mdRef").get_optional("MIMETYPE")
+        md_red_element = elem.find("METS:mdRef")
+        locref: Optional[str] = None
+        mdtype: Optional[str] = None
+        mimetype: Optional[str] = None 
+        if md_red_element:
+            locref = Utils._apply_relative_path(self.descriptor_path, md_red_element.get_optional("LOCREF"))
+            mdtype = md_red_element.get_optional("MDTYPE")
+            mimetype = md_red_element.get_optional("MIMETYPE", None)
```

Comment:
This could also maybe be simplified since `elem.find` throws errors if the element isn't found.